### PR TITLE
Fixes #1615: modal loading spinners

### LINKED ISSUE #1615: Add loader to BP
Add some loaders to the BP modal
- [x] While waiting for statistic groups to load. Could just have a spinner where they will appear.
- [x] While waiting for basin characteristics to load. Could just have a spinner where they will appear.
- Others?

### PR DESCRIPTION
**Pull request by alaws-USGS**
Closes #1615

### CODE REVIEW

**Review by harper-wavra — APPROVED** [review]
Working well, just need to resolve merge conflicts.

File: src/Controllers/BatchProcessorController.ts [hunk 1]
```diff
@@ -134,21 +140,46 @@ module StreamStats.Controllers {
...
 
         // send selected region code and retrieve flows stats list
         public getFlowStatsAndParams(rcode: string): void {
 
+            // activate spinners during state/region changes
+            if (this.flowStatsListSpinner == false || this.parametersListSpinner == false) {
+                this.flowStatsListSpinner = true;
+                this.parametersListSpinner = true;
+            }
+            console.log("before: ", this.flowStatsList)
```
Comment (amedenblik) [inline src/Controllers/BatchProcessorController.ts]: Delete console.log
```suggestion

```

File: src/Controllers/BatchProcessorController.ts [hunk 2]
```diff
@@ -134,21 +140,46 @@ module StreamStats.Controllers {
...
+            // clear flowStatsList during state/region changes
+            if (this.flowStatsList && this.flowStatsList.length > 0) {
+                this.flowStatsList.length = 0;
+            }
+            console.log("after: ", this.flowStatsList)
```
Comment (amedenblik) [inline src/Controllers/BatchProcessorController.ts]: Delete console.log
```suggestion

```

File: src/Views/batchprocessor.html [hunk 3]
```diff
@@ -105,25 +108,27 @@ <h3>Select Flow Statistics</h3>
...
-                        <td>
-                            <checkbox large ng-model="flowStat.checked" ng-change="vm.setRegionStats(flowStat)">
-                            </checkbox>
-                        </td>
-                        <td><span>&nbsp;&nbsp;{{flowStat.statisticGroupName}}</span></td>
-                    </tr>
-                </tbody>
-
-                </table>
+                <div ng-hide="!vm.flowStatsListSpinner"><i class="fa fa-spinner fa-3x fa-spin loadingSpinner"></i></div>
```
Comment (amedenblik) [inline src/Views/batchprocessor.html]: Can you please add a little space between the the blue button and the spinner?

File: src/Views/batchprocessor.html [hunk 4]
```diff
@@ -135,27 +140,31 @@ <h3>Select Basin Characteristics</h3>
...
-                                <checkbox large ng-model="parameter.checked"
-                                    ng-change="vm.updateSelectedParamList(parameter)" ng-disabled="!parameter.toggleable">
-                                </checkbox>
-                            </td>
-                            <td><span>&nbsp;&nbsp;{{parameter.code}}</span></td>
-                            <td><span>&nbsp;&nbsp;{{parameter.description}}</span></td>
-                        </tr>
-                    </tbody>
-                </table>
+                <div ng-hide="!vm.parametersListSpinner"><i class="fa fa-spinner fa-3x fa-spin loadingSpinner"></i></div>
```
Comment (amedenblik) [inline src/Views/batchprocessor.html]: Can you please add a little space between the the blue button and the spinner?

**Review by amedenblik — CHANGES_REQUESTED** [review]
Looks good! Made some small comments.

**Review by amedenblik — CHANGES_REQUESTED** [review]
Looks good! Made some small comments. Please also resolve merge conflict.

**Review by amedenblik — CHANGES_REQUESTED** [review]
Looks good! Made some small comments.

**Review by amedenblik — CHANGES_REQUESTED** [review]
Looks good! Made some small comments.

File: src/Views/batchprocessor.html (hunk 3 above)
Comment (alaws-USGS) [reply src/Views/batchprocessor.html]: Done!

File: src/Views/batchprocessor.html (hunk 4 above)
Comment (alaws-USGS) [reply src/Views/batchprocessor.html]: Done!

**Review by amedenblik — APPROVED** [review]
Looks good!

**Comment by amedenblik** [comment]
Deployed to https://dev.streamstats.usgs.gov/bp/

**Pull request merged**